### PR TITLE
Scireum/feature/smo/junit5 2

### PR DESCRIPTION
**Pull request by MOOOOOSER**
migrated URLBuilder to kotlin

### CODE REVIEW

File: src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+/**
+ * Tests the [URLBuilder] class.
+ */
+class URLBuilderTest {
+
+    @Test
+    fun `baseURL is handled correctly`() {
+
+        var urlBuilder = URLBuilder("http://sirius-lib.net")
```
Comment (fhaScireum) [inline src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt:23]:
```suggestion
    fun `baseURL is handled correctly`() {
        var urlBuilder = URLBuilder("http://sirius-lib.net")
```

File: src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        assertEquals("localhost:80", urlBuilder.build())
+
+        urlBuilder = URLBuilder("127.0.0.1:80/")
+        assertEquals("127.0.0.1:80", urlBuilder.build())
+    }
+
+    @Test
+    fun `baseURL creation with protocol and host constructor is working`() {
+
+        var urlBuilder = URLBuilder(URLBuilder.PROTOCOL_HTTP, "sirius-lib.net")
```
Comment (fhaScireum) [inline src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt:66]:
```suggestion
    fun `baseURL creation with protocol and host constructor is working`() {
        var urlBuilder = URLBuilder(URLBuilder.PROTOCOL_HTTP, "sirius-lib.net")
```

File: src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    @Test
+    fun `multiple parameters are added correctly`() {
+        val urlBuilder = URLBuilder(URLBuilder.PROTOCOL_HTTP, "sirius-lib.net")
+
+        urlBuilder.addParameter("test1", "value1")
+        urlBuilder.addParameter("test2", "value2")
+        urlBuilder.addParameter("test3", "value3")
+        assertEquals("http://sirius-lib.net?test1=value1&test2=value2&test3=value3", urlBuilder.build())
+
+    }
```
Comment (fhaScireum) [inline src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt:192]:
```suggestion
        assertEquals("http://sirius-lib.net?test1=value1&test2=value2&test3=value3", urlBuilder.build())
    }
```

File: src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+    @Test
+    fun `can't add parts after a parameter has been added`() {
+        assertThrows<IllegalStateException> {
+            val urlBuilder = URLBuilder(URLBuilder.PROTOCOL_HTTP, "sirius-lib.net")
+            urlBuilder.addParameter("test", "value")
+            urlBuilder.addSafePart("late")
+        }
+
+    }
```
Comment (fhaScireum) [inline src/test/kotlin/sirius/kernel/commons/URLBuilderTest.kt:202]:
```suggestion
        }
    }
```

**Review by mko-sci — APPROVED** [review]

**Review by idlira — APPROVED** [review]
Would be nice if we can better visualize the change in GitHub by splitting the commits in 2 parts: 1 moves the groovy class into the kotlin one, without code changes. The next adapts the syntax.

And next time, update the PR description? Helps a lot when browsing PRs if it describes what it does instead of `Scireum/feature/smo/junit5 2` 🙂

**Pull request merged**